server/src/main/java/com/cloud/server/StatsCollector.java

```diff
@@ -1599,10 +1621,21 @@ public String getConfigComponentName() {
 
     @Override
     public ConfigKey<?>[] getConfigKeys() {
-        return new ConfigKey<?>[] {vmDiskStatsInterval, vmDiskStatsIntervalMin, vmNetworkStatsInterval, vmNetworkStatsIntervalMin, StatsTimeout, statsOutputUri};
+        return new ConfigKey<?>[] {vmDiskStatsInterval, vmDiskStatsIntervalMin, vmNetworkStatsInterval, vmNetworkStatsIntervalMin, StatsTimeout, statsOutputUri, secondaryStorageCapacityThreshold};
     }
 
     public double getImageStoreCapacityThreshold() {
-        return _imageStoreCapacityThreshold;
+        double thresholdConfig = secondaryStorageCapacityThreshold.value();
+
+        if (thresholdConfig >= MIN_STORAGE_SECONDARY_CAPACITY_THRESHOLD && thresholdConfig <= MAX_STORAGE_SECONDARY_CAPACITY_THRESHOLD) {
+            return thresholdConfig;
+        }
+
+        s_logger.warn(String.format("Invalid [%s] configuration: value set [%s] is [%s]. Assuming %s as secondary storage capacity threshold.",
+                                    secondaryStorageCapacityThreshold.key(),
+                                    thresholdConfig,
+                                    thresholdConfig < MIN_STORAGE_SECONDARY_CAPACITY_THRESHOLD ? String.format("lower than '%s'", MIN_STORAGE_SECONDARY_CAPACITY_THRESHOLD) : String.format("bigger than '%s'", MAX_STORAGE_SECONDARY_CAPACITY_THRESHOLD),
+                                    MAX_STORAGE_SECONDARY_CAPACITY_THRESHOLD));
+        return MAX_STORAGE_SECONDARY_CAPACITY_THRESHOLD;
```

Comment:
i think we should revert to default, i.e. `0.9`

Reply:
I thought about it too and came to a possible resolution:

- When the config is bigger than max, I think we should return max (as it is top limit);
- When the config is lower than min, then we use default;

What do you think?

Reply:
sound like a plan

Reply:
@GutoVeronezi these value limits are controlled/validated in the configuration manager. you can add this new config to '_weightBasedParametersForValidation_' in configuration manager. check here:  https://github.com/apache/cloudstack/blob/a0d9acec6f3926ef31430a2be0c2737b30eb5ebb/server/src/main/java/com/cloud/configuration/ConfigurationManagerImpl.java#L479

Reply:
Done.